src/libraries/Common/src/System/Net/IPv6AddressHelper.Common.cs

```diff
@@ -165,18 +185,21 @@ internal static unsafe bool IsValidStrict(char* name, int start, ref int end)
 
                             // If there's more after the closing bracket, it must be a port.
                             // We don't use the port, but we still validate it.
-                            if (i + 1 < end && name[i + 1] != ':')
+                            if (i + 1 < end && name[i + 1] != TChar.CreateTruncating(':'))
                             {
                                 return false;
                             }
 
                             // If there is a port, it must either be a hexadecimal or decimal number.
-                            if (i + 3 < end && name[i + 2] == '0' && name[i + 3] == 'x')
+                            // If the next two characters are '0x' or '0X' then it's a hexadecimal number. Skip the prefix.
+                            if (i + 3 < end && name[i + 2] == TChar.CreateTruncating('0') && (name[i + 3] == TChar.CreateTruncating('x') || name[i + 3] == TChar.CreateTruncating('X')))
```

Comment:
The previous version wasn't checking for upper case X... why is this?

Reply:
It looked like a bug in the original. I can preserve that bug, but when I checked various standards (C, C# and Python) they confirmed that a hex integer literal can have a prefix of `0x` or `0X`.

Reply:
Thanks, but please revert any changes that aren't necessary just for implementing the new interface. If there's a bug to be fixed or a perf issue to be addressed, it can and should be done in a separate PR. Thanks.